proxy/proxy.go

```diff
@@ -304,6 +307,10 @@ func containerShouldAttach(container *docker.Container) bool {
 	return len(container.Config.Entrypoint) > 0 && container.Config.Entrypoint[0] == weaveWaitEntrypoint[0]
 }
 
+func containerIsWeaveRouter(container *docker.Container) bool {
+	return strings.HasPrefix(container.Config.Image, "weaveworks/weave")
```

Comment:
Not it isn't.  `container.Image` is a hex string; `container.Config.Image` is the image name it was started with.  But your point stands.